net/tcp/tcp.h

```diff
@@ -198,6 +209,7 @@ struct tcp_conn_s
 #else
   uint16_t tx_unacked;    /* Number bytes sent but not yet ACKed */
 #endif
+  uint16_t flags;         /* Flags of TCP-specific options */
```

Comment:
bool wscale;
and remove TCP_WSCALE

Reply:
The flags filed will be extended by TCP options and reused in the future, such as Selective-ACK, Time-Stamp, etc.

Reply:
If so, let's merge keepalive here too